src/include/util/set_util.h

```diff
@@ -0,0 +1,63 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// set_util.h
+//
+// Identification: src/include/util/set_util.h
+//
+// Copyright (c) 2015-18, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <set>
+
+namespace peloton {
+
+class SetUtil {
+ public:
+  // Returns if the two sets given are disjoint.
+  template <typename T>
+  static bool isDisjoint(const std::set<T> &setA, const std::set<T> &setB) {
+    bool disjoint = true;
+    if (setA.empty() || setB.empty()) return disjoint;
+
+    typename std::set<T>::const_iterator setA_it = setA.begin();
+    typename std::set<T>::const_iterator setB_it = setB.begin();
+    typename std::set<T>::const_iterator setA_end = setA.end();
+    typename std::set<T>::const_iterator setB_end = setB.end();
+
+    // make sure that the input sets are sorted
+    // if not, create a new sorted set (ascending order)
+    std::set<T> setA_sorted;
+    if (!std::is_sorted(setA.begin(), setA.end())) {
+      std::copy(setA.begin(), setA.end(),
+                std::inserter(setA_sorted, setA_sorted.begin()));
+      setA_it = setA_sorted.begin();
+      setA_end = setA_sorted.end();
+    }
+    std::set<T> setB_sorted;
+    if (!std::is_sorted(setB.begin(), setB.end())) {
+      std::copy(setB.begin(), setB.end(),
+                std::inserter(setB_sorted, setB_sorted.begin()));
+      setB_it = setB_sorted.begin();
+      setB_end = setB_sorted.end();
+    }
+
+    while (setA_it != setA_end && setB_it != setB_end && disjoint) {
+      if (*setA_it == *setB_it) {
+        disjoint = false;
+      } else if (*setA_it < *setB_it) {
+        setA_it++;
+      } else {
+        setB_it++;
+      }
+    }
```

Comment:
1. Rename to `SetUtil::IsDisjoint()`
2. Move implementation outside class definition as per [style guide](https://github.com/cmu-db/peloton/blob/master/script/coding_style.md).
3. Is the sortedness check required? `std::set` provides ordered iteration (ascending by default).

Reply:
> Is the sortedness check required? std::set provides ordered iteration (ascending by default).

Yeah, true. I did the same thing before, i.e., didn't sort it. But changed the behavior after @malin1993ml 's comment above thinking that the input set could have been initialized by the user with a different comparator. But now when I think about it again, it's impossible since the input params to this function are only set<T> and not set<T, CmpFunc>. I'll remove the superfluous sorting operation again.

Reply:
Sounds reasonable! I didn't know that `std::set` comes with default ascending order.